pennylane/gradients/parameter_shift.py

```diff
@@ -558,115 +567,119 @@ def _process_pda2_involutory(tape, pdA2, var_idx, non_involutory):
     """
     new_pdA2 = []
     for i in range(len(tape.measurements)):
-        if i in var_idx:
-            obs_involutory = i not in non_involutory
-            if obs_involutory:
-                num_params = len(tape.trainable_params)
-                item = tuple(np.array(0) for _ in range(num_params))
+
+        if i in involutory_indices:
+            num_params = len(tape.trainable_params)
+            if num_params == 1:
+                item = qml.math.array(0)

```

Comment:
Is it necessary to create an array here? It seems that later we only check equality with 0.

Reply:
Yes, it is. This is a case where `pdA2` is not `0`, but components of the gradient may be.